tests/docker_cluster.py

```diff
@@ -300,14 +300,30 @@ def exec_cmd_on_container(self, host, cmd, raise_error=True, tty=False):
         return output
 
     @staticmethod
-    def start_presto_cluster():
-        presto_cluster = DockerCluster('master',
-                                       ['slave1', 'slave2', 'slave3'],
-                                       DEFAULT_LOCAL_MOUNT_POINT,
-                                       DEFAULT_DOCKER_MOUNT_POINT)
-        presto_cluster.start_containers(INSTALLED_PRESTO_TEST_MASTER_IMAGE,
-                                        INSTALLED_PRESTO_TEST_SLAVE_IMAGE)
-        return presto_cluster
+    def start_presto_cluster(install_func):
+        if DockerCluster._check_for_presto_images():
+            presto_cluster = DockerCluster('master',
+                                           ['slave1', 'slave2', 'slave3'],
+                                           DEFAULT_LOCAL_MOUNT_POINT,
+                                           DEFAULT_DOCKER_MOUNT_POINT)
+            presto_cluster.start_containers(INSTALLED_PRESTO_TEST_MASTER_IMAGE,
+                                            INSTALLED_PRESTO_TEST_SLAVE_IMAGE)
+            return presto_cluster
+        elif install_func:
+            presto_cluster = DockerCluster.start_centos_cluster()
+            install_func(presto_cluster)
+            presto_cluster.client.commit(
+                presto_cluster.master,
+                INSTALLED_PRESTO_TEST_MASTER_IMAGE
+            )
+            presto_cluster.client.commit(
+                presto_cluster.slaves[0],
+                INSTALLED_PRESTO_TEST_SLAVE_IMAGE
+            )
+            return presto_cluster
+        else:
```

Comment:
Could we ever get to this else? install_func is a required argument so if you call it with no arguments an exception will be raised.